src/Fixer/Operator/PreIncrementFixer.php

```diff
@@ -12,107 +12,36 @@
 
 namespace PhpCsFixer\Fixer\Operator;
 
-use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\AbstractProxyFixer;
 use PhpCsFixer\FixerDefinition\CodeSample;
 use PhpCsFixer\FixerDefinition\FixerDefinition;
-use PhpCsFixer\Tokenizer\Tokens;
-use PhpCsFixer\Tokenizer\TokensAnalyzer;
 
 /**
  * @author Gregor Harlan <gharlan@web.de>
+ *
+ * @deprecated in 2.8, proxy to IncrementStyleFixer
```

Comment:
So. When we deprecate an option, method, class, we trigger `E_USER_DEPRECATION`.
Here, it's more tricky, as whole fixer got deprecated.

Yet, it could be nice to trigger `E_USER_DEPRECATION` as well, even if it won't be just few lines addition (we need to handle those deprecations!).
Why? It would prevent having both, `pre_increment` and `increment_style` in `@Symfony` ruleset, as now we need to remember (which is always hard and we will forget, like this time) to replace deprecated rule in all rulesets.

Reply:
I based deprecation on `HashToSlashCommentFixer` - there is no `E_USER_DEPRECATION`. I'd say it's good to keep both now in `@Symfony`, so when people see in README they see both belongs to `@Symfony` (some may be thinking why old isn't in there and the suggested replacement is).

I was thinking that removing whole fixer is much more easier than removing option - you remove fixer class and make sure all tests are passing (which implicates removing tests for the fixer and rulesets entries).

About not forgetting I see simple solution: there is milestone for 3.0, why not creating an issue to remove deprecated fixers now?

To add `E_USER_DEPRECATION` wouldn't it be only to trigger it in `createProxyFixer`?

Reply:
> I'd say it's good to keep both now in `@Symfony`

No, we do not keep usage of deprecated things, regardless they are methods, classes, rules.
Also, with having both of them, you end up running 2 different fixers for applying same rule, this is a CPU waste.

> so when people see in README they see both belongs to `@Symfony` (some may be thinking why old isn't in there and the suggested replacement is).

I don't believe it's a case. If sth is deprecated, old is deprecated, what does it mean? that it shall not be in use. If we ask end-user to not use it, showing we are still using it is confusing.

> I was thinking that removing whole fixer is much more easier than removing option (...)

it is, yet I don't find it related to triggering deprecation for end-user

> About not forgetting I see simple solution: there is milestone for 3.0, why not creating an issue to remove deprecated fixers now?

you are talking here about not forgetting of removing it at line 3.0, I'm saying about stop using it still on 2.x

> To add `E_USER_DEPRECATION` wouldn't it be only to trigger it in `createProxyFixer` ?

Or constructor directly.

Reply:
> No, we do not keep usage of deprecated things

Yeah, that make more sense, fixed.

> Or constructor directly.

Done, but I need a [hint](https://travis-ci.org/FriendsOfPHP/PHP-CS-Fixer/jobs/282452381).

Reply:
moving it to `applyFix` is actually not helping in initial concern I raised - forbidding to use deprecated fixers in rulesets.

Reply:
Do you mean like an adding:

```php
    // tests/RuleSetTest.php
    /**
     * @group legacy
     * @expectedDeprecation Fixer "pre_increment" is deprecated and will be removed in 3.0. Use "increment_style" instead.
     */
    public function testUsingDeprecatedTriggerError()
    {
        RuleSet::create([
            '@PSR2' => true,
            'line_ending' => true,
            'strict_comparison' => true,
        ]);
    }
```
?

This won't pass with `E_USER_DEPRECATED` triggered in constructor either.

I could move it from `applyFix` to `isCandidate` so it would be tiggered when fixer would be in use even without actual fixing.

Reply:
sth like this, yet you have no `pre_increment` in creater ruleset ;)

i think throwing deprecation from random places to find best one is not a way we shall go. 
For fixer, deprecation shall be triggered in constructor or nowhere.

Reply:
Can it be nowhere in this PR?

If I add deprecation error in constructor all tests that are using `FixerFactory::registerBuiltInFixers` (7 files) will have to be modified and also that won't be enough as still many deprecation errors will be thrown:

```
Fixer "pre_increment" is deprecated and will be removed in 3.0. Use "increment_style" instead: 203x
    153x in DescribeCommandTest::testDescribeCommand from PhpCsFixer\Tests\AutoReview
    16x in TextDiffTest::testDiffReportingDecorated from PhpCsFixer\Tests
    7x in DescribeCommandTest::setUp from PhpCsFixer\Tests\Console\Command
    6x in ConfigurationResolverTest::testConfigFinderIsOverridden from PhpCsFixer\Tests\Console
    2x in ProjectFixerConfigurationTest::testCreate from PhpCsFixer\Tests\AutoReview
    2x in ConfigTest::testConfigRulesUsingSeparateMethod from PhpCsFixer\Tests
    2x in ReadmeCommandTest::testIfReadmeFileIsCorrect from PhpCsFixer\Tests\Console\Command
    2x in ConfigTest::testCustomConfig from PhpCsFixer\Tests
    2x in ConfigTest::testConfigRulesUsingJsonMethod from PhpCsFixer\Tests
    2x in FixCommandTest::testEmptyRulesValue from PhpCsFixer\Tests\Console\Command
    1x in ApplicationTest::testApplication from PhpCsFixer\Tests\Console
    1x in FixCommandTest::setUp from PhpCsFixer\Tests\Console\Command
    1x in CommandTest::provideCommandHasNameConstCases from PhpCsFixer\Tests\AutoReview
    1x in ConfigurationResolverTest::testResolveRulesWithConfig from PhpCsFixer\Tests\Console
    1x in ConfigurationResolverTest::testResolveRulesWithOption from PhpCsFixer\Tests\Console
    1x in ConfigurationResolverTest::testResolveRulesWithUnknownRules from PhpCsFixer\Tests\Console
    1x in ConfigurationResolverTest::testResolveRulesWithConfigAndOption from PhpCsFixer\Tests\Console
    1x in ConfigurationResolverTest::testResolveCommandLineInputOverridesDefault from PhpCsFixer\Tests\Console
    1x in ConfigurationResolverTest::testResolveConfigFileOverridesDefault from PhpCsFixer\Tests\Console
```

I believe we should handle it in separate PR.

Reply:
very well, let it be

Reply:
great, I have updated PR